src/logicblocks/event/processing/broker/broker/broker_builder.py

```diff
@@ -0,0 +1,124 @@
+from abc import ABC, abstractmethod
+from datetime import timedelta
+from typing import Self, TypedDict
+
+from ..coordinator import EventSubscriptionCoordinator
+from ..locks import LockManager
+from ..nodes import NodeManager, NodeStateStore
+from ..observer import EventSubscriptionObserver
+from ..sources import (
+    EventStoreEventSourceFactory,
+    InMemoryEventSubscriptionSourceMappingStore,
+)
+from ..subscribers import (
+    EventSubscriberManager,
+    EventSubscriberStateStore,
+    InMemoryEventSubscriberStore,
+)
+from ..subscriptions import EventSubscriptionStateStore
+from .broker import CoordinatorObserverEventBroker, EventBroker
+
+
+class PrepareResults(TypedDict):
+    node_state_store: NodeStateStore
+    event_subscriber_state_store: EventSubscriberStateStore
+    event_subscription_state_store: EventSubscriptionStateStore
+    lock_manager: LockManager
+    event_source_factory: EventStoreEventSourceFactory
+
+
+class EventBrokerBuilder[**P = ...](ABC):
+    node_id: str
+
+    node_state_store: NodeStateStore
+    event_subscriber_state_store: EventSubscriberStateStore
+    event_subscription_state_store: EventSubscriptionStateStore
+    lock_manager: LockManager
+    event_source_factory: EventStoreEventSourceFactory
+
+    def __init__(self, node_id: str):
+        self.node_id = node_id
+
+    @abstractmethod
+    def _prepare(self, *args: P.args, **kwargs: P.kwargs) -> PrepareResults:
+        pass
```

Comment:
If we used `EventBrokerDependencies` instead of `PrepareResults` this method could be called `dependencies` instead. I'd also leave it public, not sure it matters for it to be "protected".

Reply:
OK!